### PR TITLE
Same channel check

### PR DESCRIPTION
**Pull request by dan-jfisher**
Quick guys, while Maarten is gone let's change everything.

I am implementing `fizyr-rpc-mi` and we need a way to check whether or not two 'Clients' are equivalent so that we can properly handle an `unsubscribe` request.  This seemed safer than exposing the underlying channel as `pub`

### CODE REVIEW

File: macros-tests/tests/camera.rs [hunk 1]
```diff
@@ -11,6 +11,13 @@ fn client_server_pair<F: fizyr_rpc::format::Format<Body = fizyr_rpc::StreamBody>
 	Ok((client.into(), server.into()))
 }
 
+#[tokio::test]
+async fn same_channel() {
+	let (client, _) = client_server_pair::<Json>().unwrap();
+	let duplicate = client.clone();
+	assert!(client.same_channel(&duplicate));
```
Comment (vcarpani) [inline macros-tests/tests/camera.rs:18]:
```suggestion
	let (client, _) = client_server_pair::<Json>().unwrap();
	let duplicate = client.clone();
	assert!(client.same_channel(&duplicate));
	
	let (new_client, _) = client_server_pair::<Json>().unwrap();
	assert!(!client.same_channel(&new_client));
```

**Review by vcarpani — COMMENTED** [review]
Changelog

**Review by ekrikelis — APPROVED** [review]
LGTM, only some minor clippy warnings

**Comment by dan-jfisher** [comment]
> LGTM, only some minor clippy warnings

Yeah those were existing warnings, I'll fix them in a seperate PR.

**Comment by dan-jfisher** [comment]
> I'm wondering if it makes sense to use `UnboundedSender::same_channel` to implement `PartialEq`?

I thought about it as well.  However, I'm hesitant to do so because `UnboundedSender` itself does not implement `PartialEq`.  I don't think I want to obscure that fact from the user.

**Comment by hgaiser** [comment]
> I thought about it as well. However, I'm hesitant to do so because `UnboundedSender` itself does not implement `PartialEq`. I don't think I want to obscure that fact from the user.

Yeah, I get your hesitance. Let's assume they have a reason for not implementing `PartialEq`? :p 

How about `same_peer` for its name though? Since `_channel` is maybe an implementation detail that we don't necessarily want to propagate.

**Comment by dan-jfisher** [comment]
> How about `same_peer` for its name though? Since `_channel` is maybe an implementation detail that we don't necessarily want to propagate.

It turns out that I actually need to using the underlying `PeerWriteHandle` in `fizyr_rpc_mi`.  Therefore, I actually don't need the macro to generate `same_peer` at all anymore.  I can either rename it or remove the autogenerated function altogether.  What do you think?

File: src/peer_handle.rs [hunk 2]
```diff
@@ -191,6 +191,11 @@ impl<Body> PeerWriteHandle<Body> {
 			command_tx: self.command_tx.clone(),
 		}
 	}
+
+	/// Check if this handle has the same underlying channel as `other`.
+	pub fn same_peer(&self, other: &Self) -> bool {
```
Comment (vcarpani) [inline src/peer_handle.rs:196]: Could you write a small unit test for this?

**Pull request merged**